crates/core/tests/fixtures.rs

```diff
@@ -0,0 +1,98 @@
+}
+
+pub enum TestContext {
+    Simple,
+    Counter,
+    PlainOpcodes,
+    ERC20,
+}
+
+impl KakarotTestEnvironmentContext {
+    pub async fn setup(test_context: TestContext) -> Self {
+        println!("async setup");
+
+        // Create a new test environment
+        let mut test_environment = KakarotTestEnvironment::new().await;
+
+        // Deploy the evm contracts depending on the test context
+        match test_context {
+            TestContext::Simple => Self { test_environment },
+
+            TestContext::Counter => {
+                // Deploy the Counter contract
+                test_environment = test_environment
+                    .deploy_evm_contract(ContractDeploymentArgs { name: "Counter".into(), constructor_args: () })
+                    .await;
+                Self { test_environment }
+            }
+            TestContext::PlainOpcodes => {
+                // Deploy the Counter contract
+                test_environment = test_environment
+                    .deploy_evm_contract(ContractDeploymentArgs { name: "Counter".into(), constructor_args: () })
+                    .await;
+                let counter = test_environment.evm_contract("Counter");
+                let counter_eth_address: Address = {
+                    let address: Felt252Wrapper = counter.addresses.eth_address.into();
+                    address.try_into().unwrap()
+                };
+
+                // Deploy the PlainOpcodes contract
+                test_environment = test_environment
+                    .deploy_evm_contract(ContractDeploymentArgs {
+                        name: "PlainOpcodes".into(),
+                        constructor_args: (EthersAddress::from(counter_eth_address.as_fixed_bytes()),),
+                    })
+                    .await;
+                Self { test_environment }
+            }
+            TestContext::ERC20 => {
+                // Deploy the ERC20 contract
+                test_environment = test_environment
+                    .deploy_evm_contract(ContractDeploymentArgs {
+                        name: "ERC20".into(),
+                        constructor_args: (
+                            Token::String("Test".into()),               // name
+                            Token::String("TT".into()),                 // symbol
+                            Token::Uint(ethers::types::U256::from(18)), // decimals
+                        ),
+                    })
+                    .await;
+                Self { test_environment }
+            }
+        }
+    }
+
+    pub fn resources(
+        &self,
+    ) -> (&KakarotTestEnvironment, &KakarotClient<JsonRpcClient<HttpTransport>>, &DeployedKakarot) {
+        (&self.test_environment, self.test_environment.client(), self.test_environment.kakarot())
+    }
+}
+
+impl Drop for KakarotTestEnvironmentContext {
+    fn drop(&mut self) {
+        block_on(async move { println!("async teardown") })
+    }
+}
```

Comment:
This is needed to drop the context after each test

Reply:
do you need to block_on?

Reply:
Finally this `Drop` is not needed, it was something I had to include when I was testing having shared test environment. Removing this 👍